prime-router/src/main/kotlin/history/azure/DeliveryFacade.kt

```diff
@@ -54,6 +62,13 @@ class DeliveryFacade(
             "End date must be after start date."
         }
 
+        var reportId: UUID?
+        try {
+            reportId = if (reportIdStr != null) UUID.fromString(reportIdStr) else null
+        } catch (e: IllegalArgumentException) {
+            throw HttpException("Invalid format for report ID: $reportIdStr", HttpStatus.UNPROCESSABLE_ENTITY)
```

Comment:
I think doing something like `return HttpUtilities.bad(request, Invalid format for report ID: $reportIdStr)` is more common (I think it's fine to return bad response rather 422, but you could also add a new helper)

Reply:
That won't work here since the return type of this method is a List and not a response.

Reply:
I updated the logic. Let me know if this is ok.